statsmodels/survey/summary_stats.py

```diff
@@ -0,0 +1,185 @@
+import numpy as np 
+import pandas as pd 
+
+class SurveyDesign(object):
+
+    def __init__(self, strata=None, cluster=None, weights=None, nest=True):
+
+    
+        # if any of them are none, we should make them an array of ones
+        # how to know the len each array should be efficiently?
+        self.weights = weights
+        # Recode strata and clusters as integer values 0, 1, ...                                                                                             
+        _, self.strat = np.unique(strata, return_inverse=True)
+        _, clust = np.unique(cluster, return_inverse=True)
+
+        # the number of distinct strata
+        self.nstrat = max(self.strat) + 1
+
+        # If requested, recode the PSUs to be sure that the same PSU number in                                                                               
+        # different strata are treated as distinct PSUs.  This is the same as                                                                                
+        # the nest option in R.                                                                                                                              
+        if nest:
+            m = max(clust) + 1
+            sclust = clust + m*self.strat
+            _, self.sclust = np.unique(sclust, return_inverse=True)
+        else:
+            self.sclust = clust.copy()
+            
+        # The number of clusters per stratum                                                                                                                 
+        _, ii = np.unique(self.sclust, return_index=True)
+        self.ncs = np.bincount(self.strat[ii])
+
+        # The stratum for each cluster                                                                                                                       
+        _, ii = np.unique(self.sclust, return_index=True)
+        self.sfclust = self.strat[ii]
+
+
+    def show(self):
+        print(self.sfclust)
+        print(self.ncs)
+        print(self.sclust)
+        print(self.nstrat)
+
+    def summary(self):
+        print("Number of observations:", len(self.strat))
+        print("Sum of weights:", self.weights.sum())
+        print("Number of strata:", self.nstrat)
+        print("The number of clusters per stratum:", self.ncs)
+
+class SurveyStat(object):
+
+    def __init__(self):
+        pass
+
+    def jack(self,stat, SurveyDesign):
```

Comment:
The design should be passed to the initializer, since it will never change.  Also, SurveyDesign is the class name, so should not also be used as an argument name.